tutorials/tour/_posts/2017-02-13-singleton-objects.md

```diff
@@ -10,65 +10,85 @@ num: 13
+class Project(name: String, daysToComplete: Int)
 
-[Scaladoc](https://wiki.scala-lang.org/display/SW/Introduction) has special support for jumping between a class and its companion: if the big “C” or “O” circle has its edge folded up at the bottom, you can click the circle to jump to the companion.
+val project1 = new Project("TPS Reports", 1)
+val project2 = new Project("Website redesign", 5)
+info("Created projects")  // Prints "INFO: Created projects"
+```
 
-A class and its companion object, if any, must be defined in the same source file. Like this:
+The `info` method becomes visible in the scope of the package using `import logging.Logger.info`. You could also use `import logging.Logger._` to import everything from Logger.
 
-```tut
-class IntPair(val x: Int, val y: Int)
+## Companion objects
 
-object IntPair {
-  import math.Ordering
+A singleton object with the same name as a class is called a _companion object_. Conversely, the class is the object's companion class. The companion class and object can access each other's private members. Use a companion object for methods and values which are not specific to instances of the companion class.
+```
+import scala.math._
 
-  implicit def ipord: Ordering[IntPair] =
-    Ordering.by(ip => (ip.x, ip.y))
+class Circle(val radius: Double) {
+  def area: Double = Circle.calculateArea(radius)
 }
-```
 
-It's common to see typeclass instances as [implicit values](implicit-parameters.html), such as `ipord` above, defined in the companion, when following the typeclass pattern. This is because the companion's members are included in the default implicit search for related values.
+object Circle {
+  def calculateArea(radius: Double): Double = Pi * pow(radius, 2.0)
+}
 
-## Notes for Java programmers ##
+val circle1 = new Circle(5.0)
 
-`static` is not a keyword in Scala. Instead, all members that would be static, including classes, should go in a singleton object instead. They can be referred to with the same syntax, imported piecemeal or as a group, and so on.
+circle1.area
+```
 
-Frequently, Java programmers define static members, perhaps `private`, as implementation aids for their instance members. These move to the companion, too; a common pattern is to import the companion object's members in the class, like so:
+The `class Circle` contains the val `radius` which is specific to each instance whereas the `object Circle` contains the method `calculateArea` which is the same for every instance.
 
+A common pattern to simplify calls to the companion object's members from the companion class is to import everything:
 ```
-class X {
-  import X._
+import java.time.LocalDateTime;
+import java.time.DayOfWeek
 
-  def blah = foo
+object Event {
+  private def _isOnTheWeekend(date: LocalDateTime) = Array(DayOfWeek.SATURDAY, DayOfWeek.SUNDAY).contains(date.getDayOfWeek)
```

Comment:
I’m not sure to see any advantage of putting this as a private def in the companion object. I think that maybe more compelling use cases for companion object’s methods are constants definition, default instances and alternative constructors.